### PR TITLE
Query and Update Animals Table

### PR DESCRIPTION
**Pull request by samiullahbahadur**
### **Project requirements**

- **_I did the below tasks:_**
- Add a column species of type string to your animals table. Modify your schema.sql file.
- **Insert the following data:**
- Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
- Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
- Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
- Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
- Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
- Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
- Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -35,4 +35,76 @@ INSERT INTO
...
+
+INSERT INTO
+  animals (
+    NAME,
+    date_of_birth,
+    escape_attempts,
+    neutered,
+    weight_kg
+  )
+VALUES ('Angemon', 'June 12, 2005', 1, TRUE, 45);
```
Comment (mmsesay) [inline data.sql]: - [x] I guess you made a mistake here. The weight for the animal `Angemon` should be a negative value `-45`

File: queries.sql [hunk 2]
```diff
@@ -31,4 +31,52 @@ WHERE NAME <> 'Gabumon';
...
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals
+
+UPDATE animals
+SET species = 'pokemon';
+COMMIT;
+SELECT * FROM animals;
```
Comment (mmsesay) [inline queries.sql:56]: - [ ] This query should not set the species for all animals to `pokemon`. As per the requirements, you should do the following:

   - UPDATE the animals table by setting the species column to digimon for all animals that have a name ending in mon.

   - UPDATE the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Commit the transaction.`

File: queries.sql [hunk 3]
```diff
@@ -31,4 +31,52 @@ WHERE NAME <> 'Gabumon';
...
+
+BEGIN
+DELETE
+FROM animals;
+ROLLBACK;
+
```
Comment (mmsesay) [inline queries.sql]: - [ ] Please add a select statement after the rollback to verify the change was rollback.

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @samiullahbahadur, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (Kingstalux) [reply data.sql]: not resolved

File: queries.sql (hunk 2 above)
Comment (Kingstalux) [reply queries.sql:56]: resolved 👍

File: queries.sql (hunk 3 above)
Comment (Kingstalux) [reply queries.sql]: resolved 👍

File: schema.sql [hunk 4]
```diff
@@ -44,4 +44,14 @@ INSERT INTO
...
+
+INSERT INTO
+  animals (
+    NAME,
+    date_of_birth,
+    escape_attempts,
+    neutered,
+    weight_kg
+  )
+VALUES ('Angemon', 'June 12, 2005', 1, TRUE, 45);
```
Comment (Kingstalux) [inline schema.sql]: - [ ] Good job so far 👍 Kindly provide the correct value for the weight of Angemon which is `-45` as requested by the previous reviewer 😄

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @samiullahbahadur, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f5754646735474252343558364e6278714a4b2f67697068792e676966


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (samiullahbahadur) [reply data.sql]: Okay, I did fix it right now

Comment (Kingstalux) [reply data.sql]: you can submit now for approval if you've corrected it

File: schema.sql (hunk 4 above)
Comment (samiullahbahadur) [reply schema.sql]: thanks , i did fix but its last chance

**Review by Kingstalux — APPROVED** [review]
Hi @samiullahbahadur,
# Status: Approved ✔️ ✔️ ✔️ 
 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966

## Highlights 👏 
✅ Great pull request description 🔥 
✅ Good commit messages 👍 
✅ Great job answering all the queries 👏 
✅ Professional README file 💯 

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**